myproject/app.py

```diff
@@ -0,0 +1,272 @@
+from flask import Flask, request, jsonify , blueprints 
+from flask_sqlalchemy import SQLAlchemy
+from flask_jwt_extended import JWTManager, create_access_token, jwt_required, get_jwt_identity
+from flask_bcrypt import Bcrypt
+from datetime import datetime
+from flask_migrate import Migrate
+from model import  db , user, post , comment , Follow , Like
+from datetime import datetime
+from utlis import add_user , new_post , delete , save_changes , add
+from config import basesit
+
+
+
+app = Flask(__name__)
+app.config.from_object(basesit)
+db.init_app(app)
+jwt = JWTManager(app)
+bcrypt = Bcrypt(app)
+migrate = Migrate(app, db)
+date = datetime.now()
+#User Management 
+@app.route("/register", methods=['POST'])
+def register_user():
+    data = request.json
+    email = data.get('email')
+    username = data.get('username')
+    password = data.get('password')
+    if not all([email, username, password]):
+        return jsonify({'error': 'All fields need to be provided'}), 400
+    if user.query.filter_by(email=email).first() or user.query.filter_by(username=username).first():
+        return jsonify({'error': 'User with this email or username already exists'}), 409
+    hashed_password = bcrypt.generate_password_hash(password).decode('utf-8')
+    add_user(email,username , hashed_password)
+    return jsonify({'message': 'User created successfully'}), 201
+
+@app.route('/login', methods=['POST'])
+def login():
+    data = request.get_json()
+    email = data.get('email')
+    password = data.get('password')
+    User = user.query.filter_by(email=email).first()
+    if not User:
+        return jsonify({'message': 'User not found'}), 404
+    if bcrypt.check_password_hash(User.password, password):
+        access_token = create_access_token(identity={'username': User.username, 'email': User.email, 'id': User.id})
+        return jsonify({'access_token': access_token}), 200
+    else:
+        return jsonify({'message': 'Invalid credentials'}), 401
+
+@app.route('/retrieve/<int:id>', methods=['GET'])
```

Comment:
If you are using the jwt decorator, why should you pass the ID inside the parameter URL?